go/teams/loader.go

```diff
@@ -517,6 +519,10 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 			l.G().Log.CDebugf(ctx, "TeamLoader fetching: NeedKeyGeneration: %v", err)
 			fetchLinksAndOrSecrets = true
 		}
+		if err := l.satisfiesNeedApplicationsAtGenerations(ctx, arg.needApplicationsAtGenerations, ret); err != nil {
+			l.G().Log.CDebugf(ctx, "TeamLoader fetching: NeedApplicationsAtGenerations: %v", err)
+			fetchLinksAndOrSecrets = true
+		}
```

Comment:
I would've thought that in order to fix the problem this `satisfiesNeedApplicationsAtGenerations` call would need to be outside of it `! hasSyncedSecrets` block. Since I thought `hasSyncedSecrets` would be true at line 925 in the test. Do you get why it works this way? Great test by the way.

Reply:
ah i think that these changes get `repoll=true` so we fetch the merkle root and see the new link with us as a member and then refetch. I'll move this outside of the conditional block though i think it makes more sense